impeller/entity/geometry/geometry.cc

```diff
@@ -119,12 +119,27 @@ GeometryResult ComputeUVGeometryForRect(Rect source_rect,
                                         RenderPass& pass) {
   auto& host_buffer = renderer.GetTransientsBuffer();
 
-  auto uv_transform =
-      texture_coverage.GetNormalizingTransform() * effect_transform;
+  // Calculate UV-specific transform based on texture coverage and effect.
+  // For example, if the texture is 100x100 and the effect transform is
+  // scaling by 2, then the UV transform should be scaling by 0.5.
+  auto uv_transform = texture_coverage.GetNormalizingTransform() *  //
+                      effect_transform;
+
+  // Allocate space for vertex and UV data (4 vertices)
+  // 0: position
+  // 1: UV
+  // 2: position
+  // 3: UV
+  // etc.
   std::vector<Point> data(8);
```

Comment:
bonus points: change this to an arrary so we don't do an extra heap allocation.

Reply:
Done!